File: equipment-service/src/main/java/edu/ngd/order/equipment/entity/Equipment.java
package edu.ngd.order.equipment.entity;

import lombok.Data;
import java.util.List;

/**
 * 设备台账实体类
 */
@Data
public class Equipment {
    /**
     * 主键
     */
    private Integer id;
    
    /**
     * 存放地点（数据字典取数）
     */
    private String storageLocation;
    
    /**
     * 设备名称
     */
    private String equipmentName;
    
    /**
     * 设备编号
     */
    private String equipmentNumber;
    
    /**
     * 产线（数据字典取数）
     */
    private String productionLine;
    
    /**
     * 状态
     */
    private String status;
    
    /**
     * 设备类型（数据字典取数）
     */
    private String deviceType;
    
    /**
     * 设备状态
     */
    private String equipmentStatus;
    
    /**
     * 责任人列表（关联关系，非数据库字段）
     */
    private List<Responsible> responsibleList;
}

File: src/types/equipment.ts
// 责任人类型
export interface Responsible {
  id?: number
  name: string
  position: string
  contact: string
  responsibleType: string
  notes: string
  equipmentLedger: number
}

// 设备类型
export interface Equipment {
  id?: number
  storageLocation: string
  equipmentName: string
  equipmentNumber: string
  productionLine: string
  status: string
  deviceType: string
  equipmentStatus: string
  responsibleList?: Responsible[]
}

// 故障代码类型
export interface EquipmentFaultCode {
  id?: number
  codeDescribe: string
  status: number
  code: string
  codeRemark: string
}

// 点检项目类型
export interface CheckProject {
  id?: number
  checkId: number
  reason: string
  remark: string
  status: number
  checkProject: string
  checkContent: string
  checkResult: string
  checkStandard: string
}

// 设备点检类型
export interface EquipmentInspection {
  id?: number
  checkNumber: string
  endTime: string | Date
  startTime: string | Date
  operateStartTime: string | Date
  operateEndTime: string | Date
  equipmentName: string
  equipmentNumber: string
  deviceType: string
  checkResult: string
  checkUser: string
  enableStatus: string
  barCode: string
  status: string
  checkProjectList: CheckProject[]
}

// 保养项目类型
export interface MaintenanceItem {
  id?: number
  maintenanceItem: string
  maintenanceContent: string
  maintenanceResult: string
  remark: string
  status: number
}

// 设备保养类型
export interface EquipmentMaintenance {
  id?: number
  maintainNumber?: string
  maintenanceNumber?: string
  equipmentName: string
  equipmentNumber: string
  deviceType: string
  maintainUser?: string
  maintenancePerson?: string
  startTime?: string | Date
  maintenanceTime?: string | Date
  endTime?: string | Date
  status: string
  maintenanceCycle?: number
  enableStatus?: string
  operateStartTime?: string | Date
  operateEndTime?: string | Date
  barCode?: string
  maintainResult?: string
  maintainDept?: string
  maintenanceItemList: MaintenanceItem[]
}

// 设备维修类型
export interface EquipmentRepair {
  id?: number
  repairNumber: string
  equipmentName: string
  equipmentNumber: string
  faultType: string
  faultDescription: string
  repairPerson: string
  repairTime: string | Date
  status: string
  repairContent: string
  repairResult: string
}

// 通用结果类型
export interface Result<T> {
  code: number
  message: string
  data: T
}

// 分页请求参数
export interface PaginationParams {
  currentPage: number
  pageSize: number
}

// 分页结果类型
export interface PaginationResult<T> {
  list: T[]
  total: number
  currentPage: number
  pageSize: number
}